Release 13.3.5

camscale opened the pull request:
* Fixed a bug in teleport-cluster Helm chart causing Teleport to crash when AWS DynamoDB autoscaling is enabled. [#30841](https://github.com/gravitational/teleport/pull/30841)
* Added Teleport Assist to Web Terminal. [#30811](https://github.com/gravitational/teleport/pull/30811)
* Fixed S3 metric name for completed multipart uploads. [#30710](https://github.com/gravitational/teleport/pull/30710)
* Added the ability for `tsh` to register and enroll the `--current-device`. [#30702](https://github.com/gravitational/teleport/pull/30702)
* Fixed Review Requests to disallow reviews after request is resolved. [#30690](https://github.com/gravitational/teleport/pull/30690)
* Ensure that SSH session errors are reported to the terminal. [#30684](https://github.com/gravitational/teleport/pull/30684)
* Fixed an issue with `tsh aws ssm start-session`. [#30668](https://github.com/gravitational/teleport/pull/30668)
* Fixed an issue with the access request failing with `invalid maxDuration`. [teleport.e#2037](https://github.com/gravitational/teleport.e/pull/2037)

### Security fix

* Security improvements with possible `medium` severity DoS conditions through protocol level attacks. [#30854](https://github.com/gravitational/teleport/pull/30854)

camscale committed:
Release 13.3.5

zmb3 commented on CHANGELOG.md:
```diff
@@ -1,5 +1,17 @@
+## 13.3.5 (08/22/23)
+
+* Fixed S3 metric name for completed multipart uploads. [#30710](https://github.com/gravitational/teleport/pull/30710)
+* Added `--current-device` capabilities to `tsh`. [#30702](https://github.com/gravitational/teleport/pull/30702)
```
```suggestion
* Added the ability for `tsh` to register and enroll the `--current-device`. [#30702](https://github.com/gravitational/teleport/pull/30702)
```

jakule replied:
We also merged new Assist features to v13 today:
```suggestion
* Added `--current-device` capabilities to `tsh`. [#30702](https://github.com/gravitational/teleport/pull/30702)
* Added Teleport Assist to Web Terminal. [#30811](https://github.com/gravitational/teleport/pull/30811)
```

jakule commented on CHANGELOG.md:
```diff
@@ -1,5 +1,17 @@
+* Fixed Review Requests to disallow reviews after request is resolved. [#30690](https://github.com/gravitational/teleport/pull/30690)
+* Fixed Teleport Connect to properly show errors from the remote end. [#30684](https://github.com/gravitational/teleport/pull/30684)
+* Fixed an issue with `tsh aws ssm start-session`. [#30668](https://github.com/gravitational/teleport/pull/30668)
+
```
```suggestion
* Fix the issue with the access request failing with `invalid maxDuration`. [https://github.com/gravitational/teleport.e/pull/2037](https://github.com/gravitational/teleport.e/pull/2037)
```

greedy52 commented on CHANGELOG.md:
```diff
@@ -1,5 +1,17 @@
+* Added `--current-device` capabilities to `tsh`. [#30702](https://github.com/gravitational/teleport/pull/30702)
+* Fixed Review Requests to disallow reviews after request is resolved. [#30690](https://github.com/gravitational/teleport/pull/30690)
+* Fixed Teleport Connect to properly show errors from the remote end. [#30684](https://github.com/gravitational/teleport/pull/30684)
+* Fixed an issue with `tsh aws ssm start-session`. [#30668](https://github.com/gravitational/teleport/pull/30668)
```
It didn't work before. Maybe:
- Fixed an issue `tsh aws ssm start-session` throws an error

camscale replied:
I tried to figure out a better description for this one as I didn't know how to describe "endpoint URL mode" concisely and usefully.

jentfoo committed:
MFA Fuzzing and Json Unmarshal Panic fix

This expands the coverage of fuzzing in `lib/web/fuzz_test.go` and includes adding seeds.
From this work a new panic was discovered where a pointer is allocated then a pointer to the pointer is passed into json.Unmarshal.  It is then possible for this original pointer to remain a `nil` reference.

This pattern looks unexpected, so all cases of double pointers being passed into json.Unmarshal were changed to the more standard empty struct pointer style to avoid potential nil reference panics.

zmb3 commented:
@jentfoo did you purposely push changes to this branch? In most cases, the release commit only updates versions and the changelog. Is there any reason your fuzzing changes can't be backported in the typical manner?

jentfoo commented:
@zmb3 It's a late inclusion, from discussions with @camscale and @r0mant we thought this would be easiest

jentfoo commented on CHANGELOG.md:
```diff
@@ -1,5 +1,17 @@
+
+### Security fix
+
+* TODO(jent): Update this when security fix lands
```
```suggestion
* Improvements with possible `medium` severity DoS conditions through protocol level attacks.
```

camscale committed:
Reword changelog entries, add security fix changelog

camscale committed:
Add changelog entries for new backports after rebase
